torch_np/_util.py

```diff
@@ -82,3 +82,22 @@ def normalize_axis_tuple(axis, ndim, argname=None, allow_duplicate=False):
         else:
             raise ValueError('repeated axis')
     return axis
+
+
+def expand_shape(arr, axis):
```

Comment:
you probably just want to pass `shape` and do `out_ndim = len(axis) + len(shape)`.

Reply:
Also, if this function is always followed by a reshape, you probably just want to call it `expand` and perform the `view` (`reshape` may copy in PyTorch`) there.
See also https://pytorch.org/docs/stable/generated/torch.Tensor.expand.html

Reply:
I think I'd keep it as a utility with shapes only (thanks!) since it is used on both tensors and wrapper ndarrays. Not sure I follow the `reshape` vs `view`, could you elaborate? 
Are you saying there is an issue with `expand_dims`, https://github.com/Quansight-Labs/numpy_pytorch_interop/pull/8/files#diff-6fff793c1370811affecb35fe3002da4f32e90d7ccf1dd2eac920c5ad86c0e02R437

Reply:
ISTM numpy.reshape semantics is also to return a view if possible, a copy otherwise:

> This will be a new view object if possible; otherwise, it will be a copy. Note there is no guarantee of the memory layout (C- or Fortran- contiguous) of the returned array. 

https://numpy.org/doc/stable/reference/generated/numpy.reshape.html

I would be *very* surprised if numpy is ever raising on a reshape as long as the number of array elements is the same.

Or are you saying that the last line of

```
def expand_dims(a, axis):
    a = asarray(a)
    shape = _util.expand_shape(a.shape, axis)
    return a.reshape(shape)
```

should read `return a.view(shape)` instead or `return a.reshape(shape)`?

Reply:
Yes, basically to signal that it will never copy. We wouldn't want this function to copy, as we're just adding a few dimensions with legnth 1.

Reply:
OK, done: https://github.com/Quansight-Labs/numpy_pytorch_interop/pull/8/commits/2da3136b2835b9d6a6016b30dfb154c016fe0776

In numpy-speak `ndarray.view` is IIUC is only about dtypes not shapes, which is why it took to this long to get to what you were saying. Anyway, as long as the main consumer here is pytorch itself., let's make it easier for pytorch :-).

Reply:
Oh, I see now. Sorry, my NumPy is not as good as it could be, and this starts to feel like talking about the differences between C and C++...